examples/src/bin/wifi_access_point_with_sta.rs

```diff
@@ -58,17 +53,16 @@ fn main() -> ! {
 
     let mut rng = Rng::new(peripherals.RNG);
 
-    let init = init(timg0.timer0, rng.clone(), peripherals.RADIO_CLK).unwrap();
+    let esp_wifi_ctrl = init(timg0.timer0, rng.clone(), peripherals.RADIO_CLK).unwrap();
 
-    let wifi = peripherals.WIFI;
+    let (mut controller, interfaces) =
+        esp_wifi::wifi::new(&esp_wifi_ctrl, peripherals.WIFI).unwrap();
 
-    let ApStaInterface {
-        ap_interface,
-        sta_interface,
-        ap_device,
-        sta_device,
-        mut controller,
-    } = create_ap_sta_network_interface(&init, wifi).unwrap();
+    let mut ap_device = interfaces.ap;
+    let ap_interface = ap_device.interface();
```

Comment:
My only note here with the API is this would've been nice as a one liner like so.
```Rust
let (mut ap_device, ap_interface) = interfaces.ap;
```

But I'm not sure how nicely this plays with smoltcp being an optional feature. From reading the code I'm struggling to draw a line between esp-wifi and smoltcp, so idk what's sensible to do here.

Reply:
I agree that we are probably integrating smoltcp a bit more tightly than ideal. Since it's at 0.12 or so at the moment, it'll bite us in the future.

Reply:
I decided to remove that completely - for now users can do it themselves and mid-term/long-term we should come up with a better plan for the smoltcp and embassy-net-driver dependencies

Also we want to push users to use async 🤣